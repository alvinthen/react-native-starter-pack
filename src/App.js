// @flow
import React from 'react';
import { ActivityIndicator } from 'react-native';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import NavigatorWithState from './Navigator';
import createStore from './redux/createStore';

const { persistor, store } = createStore();

const App = () => (
  <Provider store={store}>
    <PersistGate
      loading={<ActivityIndicator />}
      persistor={persistor}
    >
      <NavigatorWithState />
    </PersistGate>
  </Provider>
);

export default App;
