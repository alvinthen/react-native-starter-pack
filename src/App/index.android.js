import React from 'react';
import {
  Text,
  View,
} from 'react-native';
import styles from './style';

/* eslint-disable react/prefer-stateless-function */
export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit index.android.js
        </Text>
        <Text style={styles.instructions}>
          Double tap R on your keyboard to reload,{'\n'}
          Shake or press menu button for dev menu
        </Text>
      </View>
    );
  }
}
