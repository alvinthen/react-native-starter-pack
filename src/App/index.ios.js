import React from 'react';
import {
  Text,
  View,
} from 'react-native';
import styles from './style';

/* eslint-disable react/prefer-stateless-function */
export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit index.ios.js
        </Text>
        <Text style={styles.instructions}>
          Press Cmd+R to reload,{'\n'}
          Cmd+D or shake for dev menu
        </Text>
      </View>
    );
  }
}
