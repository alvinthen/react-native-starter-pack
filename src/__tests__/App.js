import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
