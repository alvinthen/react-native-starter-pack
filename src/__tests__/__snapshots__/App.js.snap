// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
/>
`;
